### ISSUE TITLE
'Add Mode of Inheritance' button in the VCI

### ISSUE THREAD

**Issue by wrightmw** [ClinGen/clincoded#990]
![screen shot 2016-09-09 at 5 04 33 pm](https://cloud.githubusercontent.com/assets/15131169/18406250/8b8831ac-76af-11e6-935f-31894b8a8452.png)

An 'Add MOI +' button should appear in all pages of the Interpretation Record, as shown in the figure above.  

Note: the MOI button should only appear once a Disease has been selected. 

Once selected a pull-down of the different MOIs should appear in a modal.

When an MOI has been selected then the button should change from 'Add MOI +' to 'Edit MOI +', and the mode of inheritance should appear below the title.

![screen shot 2016-09-09 at 5 18 40 pm](https://cloud.githubusercontent.com/assets/15131169/18406426/944996da-76b1-11e6-9422-5df18cca376c.png)

Additonally, the current "Associate with Disease' button should be renamed as 'Add Disease +'.

If 'Add MOI +' and 'Add Disease +' take up too much room then perhaps try ''MOI +' and 'Disease +'

**Comment by mrmin123** [comment]
Change buttons to [ Disease (+) ] [ Inheritance (+) ]. (+) will change to (pencil) to edit.

**Comment by mrmin123** [comment]
@selinad 
![image](https://cloud.githubusercontent.com/assets/4326866/18653545/75bbd094-7e8f-11e6-8bcd-185286407617.png)

Adding Inheritance (new Oligogenic option added):
![image](https://cloud.githubusercontent.com/assets/4326866/18653564/882d46e0-7e8f-11e6-867a-be44289f0fc2.png)

Inheritance Added:
![image](https://cloud.githubusercontent.com/assets/4326866/18653576/a28905f6-7e8f-11e6-8c4b-d6e8575a13c2.png)

Disease and Inheritance Added:
![image](https://cloud.githubusercontent.com/assets/4326866/18653584/b2575a78-7e8f-11e6-9602-084063849852.png)

Note: Mode of Inheritance does not show up anywhere on the VCI

**Comment by mrmin123** [comment]
@selinad instance should be up on http://990-mc-moi-modal-3894506-minchoi.instance.clinicalgenome.org/ in about 20-30 minutes, if all goes well.

**Comment by selinad** [comment]
Looking sweet!  I'm probably too late (sorry, I somehow missed your mail), but I think disease on left, mode on right might be better.  The functionality is most important, so don't change it if in any way time consuming.  thank you!

**Comment by mrmin123** [comment]
@selinad no problem! Instance was updated here: http://990-mc-moi-modal-3894506-minchoi.instance.clinicalgenome.org/

Also has non-funky history items on the Dashboard:

![image](https://cloud.githubusercontent.com/assets/4326866/18654570/256379e6-7e97-11e6-8b31-7fdcc7b18dfd.png)

**Comment by selinad** [comment]
Hi @mrmin123 - thx!

Buttons look great. "+" and pencil icons super.  Mode of inheritance with oligogenic also looks good.

Only a couple of issues:
- unable to remove a disease when editing - says it's a required field and does the spinning thing (wasn't this fixed previously?)
<img width="590" alt="screen shot 2016-09-19 at 10 53 57 pm" src="https://cloud.githubusercontent.com/assets/11320314/18658947/055ff870-7ebc-11e6-9778-1b55dbbfe2b3.png">

- similar for mode of inheritance - won't let me edit to "Select"

- can mode of inheritance appear under disease at top? (like for GCI)
<img width="488" alt="screen shot 2016-09-19 at 10 55 34 pm" src="https://cloud.githubusercontent.com/assets/11320314/18658961/3748d14a-7ebc-11e6-85ae-38643d97c23f.png">

- this instance says "View Evidence" instead of "Evidence View" - was it before the change?  

(I assume this is not yet integrated with Kang's static bar so I should not be checking that, but let m know if otherwise)

thx @mrmin123 !

**Comment by selinad** [comment]
p.s. what's a non-funky history?!  (what you show looks great!).

**Comment by mrmin123** [comment]
@selinad No, I don't think those were fixed previously, judging by the behavior's existence on other instances... I can certainly fix the spinner spinning on empty field, but the ability to remove the disease probably won't go in this release. Same for the MOI.

Also, what would you like the text to say for the inheritance at the top?

**Comment by selinad** [comment]
Hi @mrmin123 I think it was #945 - it was a related issue.  

It's a problem if we can't allow someone to delete a disease or moi, unfortunately.  We will need to supply a warning note:

"Note: In this version, once you have associated a disease with an interpretation, you may edit the disease but not remove a disease association from an interpretation."

(or something like that - edits welcome).

wrt to moi text, are you talking about at the top of the modal?  thx!

**Comment by mrmin123** [comment]
@selinad I meant for this:

- can mode of inheritance appear under disease at top? (like for GCI)
<img width="488" alt="screen shot 2016-09-19 at 10 55 34 pm" src="https://cloud.githubusercontent.com/assets/11320314/18658961/3748d14a-7ebc-11e6-85ae-38643d97c23f.png">

**Comment by selinad** [comment]
Got it - sorry!  I was thinking like the GDM:

<img width="425" alt="screen shot 2016-09-20 at 10 44 08 am" src="https://cloud.githubusercontent.com/assets/11320314/18681977/45cdb6d2-7f1f-11e6-94d9-ba4c1867129c.png">

But, we have the disease next to gene there.  Could we put it in parentheses in italics and remove a few words?
"This interpretation associated with **Achondroplasia** (_autosomal dominant inheritance_)"

**Comment by selinad** [comment]
Also in this same are when there is no interpretation, it says "Evidence View Only" - could this be changed to "Evidence View?"  (no biggie if not, or let me know if I should include that item somewhere else)

**Comment by mrmin123** [comment]
@selinad what if it's associated with no disease, but with a MOI?

**Comment by selinad** [comment]
that would be odd curation, but good point -  how about:

This interpretation is associated with **Achondroplasia** -  _autosomal dominant inheritance_

and

This interpretation is associated with **no disease** -  _autosomal dominant inheritance_

**Comment by mrmin123** [comment]
@selinad updated instance (please refresh): http://990-mc-moi-modal-3894506-minchoi.instance.clinicalgenome.org/

* Fixed the text 'View Evidence' to 'Evidence View'
* Fixed the text 'Evidence View only' to 'Evidence View'
* Revised sub-title text to display MOI
* Can delete Disease and MOI now

**Comment by selinad** [comment]
Hi @mrmin123 - fantastic!!!  Thank you so much for making it possible to deleted a disease or moi - that's a really big thing, especially after #945 

Everything works/looks great.  2 very small changes:
- if quick, remove display of HPO code next to moi on the title
- could you please make this text NOT in italics: "This interpretation is associated with..." - that way, the disease will be bold and moi in italics, which I think will help make the moi stand out better.

I couldn't find a disease and moi string that would be too long - I assume it will just wrap...?

thx again!  -S

**Comment by wrightmw** [comment]
@mrmin123 This looks great! I think the curators only wanted to be able to click the Inheritance button once a Disease has been associated. I can add this as a ticket for a future release.

**Comment by mrmin123** [comment]
@wrightmw lets move that logic to a separate ticket.

Update title:
![image](https://cloud.githubusercontent.com/assets/4326866/18691524/ff5d7a3c-7f48-11e6-98bf-b9f1a8df728a.png)

History for removing disease/moi from interpretation:
![image](https://cloud.githubusercontent.com/assets/4326866/18691502/dca8262c-7f48-11e6-91c7-999b00bf9e67.png)

**Comment by mrmin123** [comment]
@selinad @wrightmw new instance with latest changes here: http://990-mc-moi-modal-ffafa52-minchoi.instance.clinicalgenome.org/

**Comment by selinad** [comment]
@mrmin123 - it's fantastic!!!  I can edit, disassociate disease or moi, all icons, buttons are good, can go in and out of the interpretation and all is well.  Titles are accurate

History great too!

Tiniest of tiniest if quick, for consistency - for the GCI, the moi is in italics on the history - can we do this for the VCI as well?  Right now the moi is in bold:

![screen shot 2016-09-20 at 4 33 27 pm](https://cloud.githubusercontent.com/assets/11320314/18692573/0d240404-7f50-11e6-9a06-b5451b13b025.png)

Again, this is not required for this release ....

thx!  Great work!

**Comment by wrightmw** [comment]
@mrmin123 I concur with @selinad ... looks great... fab work!

**Comment by selinad** [comment]
Hi @mrmin123 - moi works beautifully!  Great job!

@jimmyzhen - one thing I just noticed - if the moi is not an HPO term (e.g. codominant), the mode does not display on the Summary page:
<img width="837" alt="screen shot 2016-09-29 at 9 59 47 pm" src="https://cloud.githubusercontent.com/assets/11320314/18981177/9b64ae34-8690-11e6-8e18-23be34d2fea8.png">

Please advise - should I make new ticket?

**Comment by jimmyzhen** [comment]
Hi @selinad,

Please create a new ticket for it. I can address it. Thanks!

**Comment by selinad** [comment]
@jimmyzhen - new ticket made: #1028

**Comment by kilodalton** [comment]
Included in the last release (R7). This included the Variant Curation Interface. Nice job and thanks for your hard work.
